### DISCUSSION TITLE
Kibana/Elastic search - 2 shards have failed

### DISCUSSION THREAD

**Discussion by zitstif** [Security-Onion-Solutions/securityonion#7679] [Unsupported Versions] [answered]
Hi All! I'm new to security-onion but so far I'm finding it to be such a wonderful tool. Within the last two weeks I've noticed that nothing is showing up under 'Hunt' and out of curiosity I wanted to check the status of kibana/elastic search. A quick popup mentions there are 2 shards that have failed. How do I go about troubleshooting this? 

so-status shows everything as 'OK'.  SecurityOnion version is 2.3.91.  Attached should be the kibana log. Any help would be greatly appreciated! 

[kibana.log](https://github.com/Security-Onion-Solutions/securityonion/files/8392792/kibana.log)

**Comment by cyb3rz3us** [comment, marked as answer]
Hi @zitstif - I too am fairly new to SO so nowhere near an expert but I have also had to troubleshoot a few shard\index issues.  One thing that helped me troubleshoot these better was understanding the architecture.  With that in mind, the Elastic folks have a killer tutorial that walks one through ES including 2 sections called "Life inside a cluster" and "Inside a shard".  Also understanding node health and the index\shard state is key so I have provided some commands I use when troubleshooting.

- **Sensor log:** /opt/so/log/soc/sensoroni-server.log
- **ES docs:** https://www.elastic.co/guide/en/elasticsearch/guide/current/getting-started.html
- **Life inside a cluster:** https://www.elastic.co/guide/en/elasticsearch/guide/current/distributed-cluster.html
- **Inside a shard:** https://www.elastic.co/guide/en/elasticsearch/guide/current/inside-a-shard.html

#### Check overall node health
```sudo so-elasticsearch-query _cluster/health?pretty```
#### List index state (red, yellow, or green - you specify)
```sudo so-elasticsearch-query _cat/indices?v | grep [ red|yellow|green ]```
#### Show index state COUNT (red, yellow, or green - you specify)
```sudo so-elasticsearch-query _cat/indices?v | grep [ red|yellow|green ] | wc -l```
#### List shard state
```curl -u [ USER ]:[ PW ] -kX GET https://[ IP ]:9200/_cat/shards?pretty```
#### List unassigned shards along with reason
```curl -u [ USER ]:[ PW ] -kX GET https://[ IP ]:9200/_cat/shards?h=index,shard,prirep,state,sto,sc,unassigned.reason,unassigned.details | grep UNASS```

**Reply by zitstif** [reply]
Thanks for the reply and the insight! 

So right now there are two unassigned shards:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 17328  100 17328    0     0  58607      0 --:--:-- --:--:-- --:--:-- 58738
so-zeek-2022.01.23                                            3 p UNASSIGNED
so-zeek-2022.01.23                                            1 p UNASSIGNED

The last command results in this:

 % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 15729  100 15729    0     0  41771      0 so-zeek-2022.01.23                                            3 p UNASSIGNED            CLUSTER_RECOVERED
-so-zeek-2022.01.23                                            1 p UNASSIGNED            CLUSTER_RECOVERED
-:--:-- --:--:-- --:--:-- 41832

I've done some googling but I'm not 100% certain what I need to do from here... Do I need to reallocate the shards like this article states?

https://community.appdynamics.com/t5/Knowledge-Base/How-do-I-fix-an-Events-Service-cluster-that-is-stuck-in-red/ta-p/29188

**Reply by cyb3rz3us** [reply]
In my case, when I've had unassigned shards, I ended up deleting them.  My research indicated that because the reason is ```CLUSTER_RECOVERED```, the risk of data loss is small to nil.  Again, this is my research and it may be inaccurate so you may want to confirm for yourself or see if anyone else responds.

All that said, please use the following command **VERY CAREFULLY**...

```curl -u [ USER ]:[ PW ] -kX DELETE https://[ IP ]:9200/[ SHARD_NAME ]```

**Reply by zitstif** [reply]
Thanks for the follow up! When I try to delete the unassigned shards, I'm getting this response:

sudo curl -u username:password -kX DELETE https://127.0.0.1/9200/so-zeek-2022.01.23
<html>
<head><title>307 Temporary Redirect</title></head>
<body>
<center><h1>307 Temporary Redirect</h1></center>
<hr><center>nginx</center>
</body>
</html>

**Reply by cyb3rz3us** [reply]
Did you try also using the IP of the host?

**Reply by cyb3rz3us** [reply]
I also just noticed that your syntax is wrong - replace ```127.0.0.1/9200``` with ```127.0.0.1:9200```

**Reply by zitstif** [reply]
Oh jeeze! lol Thanks!

**Reply by zitstif** [reply]
It's all working now! Thank you so much! I hate overlooking simple mistakes like that.

**Reply by cyb3rz3us** [reply]
All good - glad you were able to get it up and running.  One thing I have learned about SO is the value of the community.  Really is a great group and an essential resource to grappling with SO and the attendant learning curve.  It is a complex beast but incredibly functional so well worth the effort...
